### PR TITLE
Fixing gas reporter for random-beacon

### PR DESCRIPTION
**Pull request by dimpar**
Refs https://github.com/keep-network/keep-core/issues/2955

Combination of overwriting the hardhat's "test" task and requiring test fixtures `require("./test/fixtures")` resulted in hiding the gas report after the tests run. Instead of overwriting a "test" task, we can also create a new task that checks the count of needed accounts and run this check before the main "test" task.

Another addition to this PR is fetching the current market data from the Coinmarketcap. The gas report will show the current price for executing specific functions and the cost of contract deployments based on the live data. The free tier API key is good for 10000 requests per month. Execution: `COINMARKETCAP_API_KEY=<key> yarn test` For testing purposes please see [8374135](https://github.com/keep-network/keep-core/pull/2958/commits/837413537b26ed3b7ac0acc2caf5d496d477a8b7) ;)

### CODE REVIEW

File: solidity/random-beacon/hardhat.config.ts [hunk 1]
```diff
@@ -133,21 +131,23 @@ const config: HardhatUserConfig = {
...
+  },
 }
 
-task(TASK_TEST, "Runs mocha tests").setAction(async (args, hre, runSuper) => {
+task("check-accounts-count", "Checks accounts count").setAction(async () => {
   // eslint-disable-next-line @typescript-eslint/no-var-requires,global-require
   const { constants } = require("./test/fixtures")
 
-  if (testConfig.operatorsCount < constants.groupSize) {
+  if (testConfig.operatorsCount < constants.groupSize4) {
```
Comment (nkuba) [inline solidity/random-beacon/hardhat.config.ts:144]:
```suggestion
  if (testConfig.operatorsCount < constants.groupSize) {
```

File: solidity/random-beacon/package.json [hunk 2]
```diff
@@ -15,7 +15,8 @@
   "scripts": {
     "clean": "hardhat clean",
     "build": "hardhat compile",
-    "test": "hardhat test",
+    "check-accounts-count": "hardhat check-accounts-count",
```
Comment (nkuba) [inline solidity/random-beacon/package.json:18]: We don't need to define this script. It's enough that we included task execution in `test` script.
```suggestion
```

**Review by nkuba — CHANGES_REQUESTED** [review]

File: solidity/random-beacon/hardhat.config.ts (hunk 1 above)
Comment (dimpar) [reply solidity/random-beacon/hardhat.config.ts]: oops, fat finger fixed in [b7d2f6e](https://github.com/keep-network/keep-core/pull/2958/commits/b7d2f6e3a86b5b3161deab98a8a2cb33c599b0f2)

File: solidity/random-beacon/package.json (hunk 2 above)
Comment (dimpar) [reply solidity/random-beacon/package.json]: right, removed in [fcbfa94](https://github.com/keep-network/keep-core/pull/2958/commits/fcbfa946e9f1c860de440c12e5eeb05a24965322)

**Review by nkuba — APPROVED** [review]

**Pull request merged**